test/qa/lib/network.py

```diff
@@ -53,10 +52,15 @@ def _capture_packets(connection_settings: (str, str, str)) -> int:
         traffic_filter = TSHARK_FILTER_TCP_OBFUSCATED % server_ip
 
     # If enough packets are captured, do not wait the duration time, exit early, show compact output
-    #tshark_result: str = sh.tshark("-i", "any", "-T", "fields", "-e", "ip.src", "-e", "ip.dst", "-a", "duration:3", "-a", "packets:1", "-f", traffic_filter)
-    tshark_result: str = os.popen(f"sudo tshark -i any -T fields -e ip.src -e ip.dst -a duration:3 -a packets:1 -f {traffic_filter}").read()
+    tshark_result: str = sh.tshark("-i", "any", "-T", "fields", "-e", "ip.src", "-e", "ip.dst", "-a", "duration:3", "-a", "packets:1", "-f", traffic_filter)
+    #tshark_result: str = os.popen(f"sudo tshark -i any -T fields -e ip.src -e ip.dst -a duration:3 -a packets:1 -f {traffic_filter}").read()
 
-    packets = tshark_result.strip().split("\n")
+    if tshark_result.strip() != "":
```

Comment:
what is the difference between from previously, because if string is empty `len` would return 0?

Reply:
It was not returning 0, it was actually returning 1, after we split string by newlines, try this:
```
a = ""
b = a.split("\n")
len(b)
```

I changed this logic in here, because if no packets were captured, everybody expects, that `len(packets)` will be 0, but it was not :smile:

Reply:
right, has different output if is using custom separator or default space :melting_face: .
you could use `splitlines()` :laughing:

Reply:
`splitlines()` seems like a better solution, will do that instead 🫡